src/frontend/lib/utils.ts

```diff
@@ -278,3 +269,47 @@ function leftPad(str: string, len: number, char: string): string {
 //   // TODO change how we determine whether we are in practice mode or not
 //   return config.metadata.name === "mapeo-default-settings";
 // }
+
+export function matchPreset(
+  availableTags: Observation['tags'],
+  presets: Preset[],
+): Preset | undefined {
+  let bestMatch: Preset | undefined;
+  let bestMatchScore = 0;
+
+  presets.forEach(preset => {
+    let score = 0;
+    let presetTagsCount = Object.keys(preset.tags).length;
+    let matchedTagsCount = 0;
+
+    for (const key in preset.tags) {
+      if (preset.tags.hasOwnProperty(key)) {
+        const presetTag = preset.tags[key];
+        const availableTag = availableTags[key];
+        if (presetTag === availableTag) {
+          score++;
+          matchedTagsCount++;
+        } else if (
+          Array.isArray(presetTag) &&
+          (presetTag as (boolean | number | string | null)[]).includes(
```

Comment:
Can remove the type casting here:

```suggestion
          presetTag.includes(
```